predict.py

```diff
@@ -1,94 +1,280 @@
+        # this signal handler passes along the cancellation to the multigpu inference server 
+        signal.signal(signal.SIGUSR1, self._get_signal_handler())
+
+        # stuff that the server needs to know
+        self.WORLD_SIZE = torch.cuda.device_count()
+        self.MODEL_CACHE = "checkpoints"
+        self.MODEL_URL = "https://weights.replicate.delivery/default/hunyuanvideo-community/HunyuanVideo/model.tar"
+        self.HTTP_PORT = 5001
+        print(f"inference server will use {self.WORLD_SIZE} devices")
+
+        if not os.path.exists(self.MODEL_CACHE):
+            self._download_weights()
+        
+        # initialize the model specific predictor and server
+        # the predictor is a wrapper around the HunyuanVideoPipeline
+        # the server is an inference server that runs the predictor
+        self.model = HunyuanPredictor(self.MODEL_CACHE, self.WORLD_SIZE, self.HTTP_PORT, self.WORLD_SIZE)
+        self.server_process = mp.Process(target=run_server, args=(self.model, self.HTTP_PORT, self.WORLD_SIZE))
+        self.server_process.start()
+        self.request_cancelled = False # internal flag that is used to manage cancellation of inference requests
         
-        self.transformer = HunyuanVideoTransformer3DModel.from_pretrained(
-            MODEL_CACHE,
-            subfolder="transformer",
-            torch_dtype=torch.bfloat16,
-        )
-        self.pipe = HunyuanVideoPipeline.from_pretrained(
-            MODEL_CACHE,
-            transformer=self.transformer,
-            torch_dtype=torch.float16,
-        )
-        self.pipe.vae.enable_tiling()
-        self.pipe.to("cuda")
+        # warmup
+        predict_args = {
+            'prompt': 'A cat walks on the grass, realistic style',
+            'width': 864,
+            'height': 480,
+            'video_length': 61,
+            'infer_steps': 1,
+            'embedded_guidance_scale': 6.0,
+            'fps': 24,
+            'seed': 42,
+            'output_path': 'warmup.mp4'
+        }
+        self._request_and_wait(predict_args, timeout=float('inf'))
+        print("Server is up and ready")
+    
 
     def predict(
         self,
         prompt: str = Input(
             description="The prompt to guide the video generation",
-            default="A cat walks on the grass, realistic",
+            default="A cat walks on the grass, realistic style",
         ),
         width: int = Input(
-            description="Width of the video in pixels", 
-            default=960, ge=512, le=1280
+            description="Width of the video in pixels (must be divisible by 16)", 
+            default=864, ge=16
         ),
         height: int = Input(
-            description="Height of the video in pixels",
-            default=544, ge=320, le=720
+            description="Height of the video in pixels (must be divisible by 16)",
+            default=480, ge=16
         ),
```

Comment:
good to keep the names of these parameters consistent between updates so as to not change the API